lib/osf-components/addon/components/addons-service/addon-account-setup/component.ts

```diff
@@ -197,20 +169,78 @@ export default class AddonAccountSetupComponent extends Component<Args> {
         }
     }
 
+    @task
+    @waitFor
+    async connectAccount() {
+        const { manager } = this.args;
+        const credentials = this.credentialsObject;
+        let apiBaseUrl;
+        if (this.showUrlField) {
+            apiBaseUrl = this.url;
+        } else if (this.showRepoOptions) {
+            apiBaseUrl = this.otherRepoSelected ? this.otherRepo : this.selectedRepo;
+        }
```

Comment:
if we can use a distinct external-service per dataverse (and similar), do we need the "repo" options at all? by the time they get here, they'll already have made this choice